PlayCover/AppInstaller/Downloader.swift

```diff
@@ -0,0 +1,121 @@
+//
+//  Downloader.swift
+//  PlayCover
+//
+//  Created by Amir Mohammadi on 9/26/1401 AP.
+//
+
+import Foundation
+import DownloadManager
+
+/// DownloaderManager can be configured through this struct, default values are as the same as below
+/// `public struct DownloadManagerConfig {`
+///    `public var maximumRetries = 3`
+///    `public var exponentialBackoffMultiplier = 10`
+///    `public var usesNotificationCenter = false`
+///    `public var showsLocalNotifications = false`
+///    `public var logVerbosity: LogVerbosity = .none
+/// `}`
+///  Use `downloader.configuration = DownloadManagerConfig()` in
+///  `DownloadApp` class before `downloader.addDownload` to apply
+///  More details: https://github.com/shapedbyiris/download-manager/blob/master/README.md
+
+class DownloadApp {
+    let url: URL
+    let app: StoreAppData
+    let warning: String?
+
+    init(url: URL, app: StoreAppData, warning: String?) {
+        self.url = url
+        self.app = app
+        self.warning = warning
+    }
+
+    let downloadVM = DownloadVM.shared
+    let installVM = InstallVM.shared
+    let downloader = DownloadManager.shared
+
+    func start() {
+        if downloadVM.downloading && installVM.installing {
```

Comment:
IMO you should add a `NetworkVM` check here

Reply:
Hmm, ok.